File: src/main/java/injea/knwremodel/Notice/Notice.java
package injea.knwremodel.Notice;


import com.fasterxml.jackson.annotation.JsonIgnore;
import injea.knwremodel.Comment.Comment;
import injea.knwremodel.Like.Like;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "notices")
@Getter
@ToString
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Notice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(updatable = false)
    private Long id;

    @Column(nullable = true)
    private Long boardId;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String major;

    @Column(nullable = false)
    private String type;

    @Column(nullable = false)
    private String writer;

    @Column(nullable = false)
    private LocalDate regdate;

    @Column(nullable = false)
    private Long view;

    @Column(columnDefinition = "TEXT", nullable = true)
    private String body;

    @Column(columnDefinition = "TEXT", nullable = true)
    private String img;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String html;
    
    @JsonIgnore
    @OneToMany(mappedBy = "notice", fetch = FetchType.EAGER, cascade = CascadeType.REMOVE)
    @OrderBy("id asc") // 댓글 정렬
    private List<Comment> comments = new ArrayList<>();

    @JsonIgnore
    @OneToMany(mappedBy = "notice", fetch = FetchType.EAGER, cascade = CascadeType.REMOVE)
    private List<Like> likes = new ArrayList<>();

    @Column(nullable = false)
    private Long likeCount;

    @Builder
    public Notice(Long boardId, String title, String type, String major, String writer,
                  LocalDate regdate, Long view, String body, String img, String html) {
        this.boardId = boardId;
        this.title = title;
        this.type = type;
        this.major = major;
        this.writer = writer;
        this.regdate = regdate;
        this.view = view;
        this.body = body;
        this.img = img;
        this.html = html;
        this.comments = getComments();
        this.likeCount = 0L;
    }

    public void updateLikeCount(Long likeCount) {
        this.likeCount = likeCount;
    }

    public LocalDate getCreateDate() {
        return regdate;
    }


}

File: src/main/kotlin/injea/knwremodel/Notice/Notice.kt
package injea.knwremodel.Notice

import com.fasterxml.jackson.annotation.JsonIgnore
import injea.knwremodel.Comment.Comment
import injea.knwremodel.Like.Like
import jakarta.persistence.*
import java.time.LocalDate


@Entity
@Table(name = "notices")
class Notice(
    boardId: Long?,
    title: String,
    major: String,
    type: String,
    writer: String,
    createDate: LocalDate,
    view: Long,
    body: String,
    img: String?,
    html: String,

    likeCount: Long = 0
){
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(updatable = false)
    private val id: Long? = null

    var boardId = boardId
        protected set
    var title = title
        protected set
    var major = major
        protected set
    var type = type
        protected set
    var writer = writer
        protected set
    var createDate = createDate
        protected set
    var view = view
        protected set

    @Column(columnDefinition = "TEXT")
    var body = body
        protected set

    @Column(columnDefinition = "TEXT")
    var img = img
        protected set

    @Column(columnDefinition = "TEXT", nullable = false)
    var html = html
        protected set

    @JsonIgnore
    @OneToMany(mappedBy = "notice", fetch = FetchType.EAGER, cascade = [CascadeType.REMOVE])
    @OrderBy("id asc") // 댓글 정렬
    val comments: MutableList<Comment> = mutableListOf()

    @JsonIgnore
    @OneToMany(mappedBy = "notice", fetch = FetchType.EAGER, cascade = [CascadeType.REMOVE])
    val likes: MutableList<Like> = mutableListOf()

    @Column(nullable = false)
    var likeCount = likeCount
        protected  set
}
